src/test/java/ejercicio/banco/repository/MySqlBankRepositoryTest.java

```diff
@@ -8,90 +8,110 @@
 import org.mockito.Mock;
 import org.mockito.runners.MockitoJUnitRunner;
 
+import java.sql.SQLException;
 import java.util.ArrayList;
 import java.util.List;
 
 import static org.junit.Assert.*;
+import static org.mockito.Mockito.when;
 
 @RunWith(MockitoJUnitRunner.class)
 public class MySqlBankRepositoryTest extends AbstractMySqlRepositoryTest {
-
-    private static final String TABLE = "bank_project_test";
-    private static final String NOTHING = "nothing";
-    private static final String EMPTYTABLE = "empty";
-    private List<Bank> banks = new ArrayList();
-    private List<Bank> empty = new ArrayList();
     private Bank bank;
     private Bank expectedBank1 = new Bank(1, "bestBank", "right here");
     private Bank expectedBank2 = new Bank(2, "Worst ATM", "over there");
-    private List<Bank> full = new ArrayList();
 
     @Mock
     private DataBaseUtil util;
 
     @InjectMocks
     MySqlBankRepository repository;
 
-    @Test
-    public void testFindAllReturnsEmptyListWhenNull() {
-        banks = repository.findAll(null);
-        assertEquals(empty, banks);
-    }
 
     @Test
-    public void testFindAllReturnsEmptyListWhenListNotFound() {
-        banks = repository.findAll(NOTHING);
-        assertEquals(empty, banks);
+    public void testFindAllReturnsFullList() throws SQLException {
+        executeQuery(util, "select * from BANKS");
+
+        when(resultSet.next()).thenReturn(true).thenReturn(false);
+        when(resultSet.getString("name")).thenReturn(expectedBank2.getName());
+        when(resultSet.getInt("id")).thenReturn(expectedBank2.getId());
+        when(resultSet.getString("address")).thenReturn(expectedBank2.getAddress());
+
+        List<Bank> banks = new ArrayList();
+        banks.add(expectedBank1);
+        banks.add(expectedBank2);
+        banks = repository.findAll("");
+        assertFalse(banks.isEmpty());
+        assertEquals(banks.size(), banks.size());
```

Comment:
![ohmygod](https://media.giphy.com/media/21GCae4djDWtP5soiY/giphy.gif)

Why? why god!?

Now speaking seriously, can you explain that code? Please review it again and tell me what you wanted to achieve.

Reply:
Upsie, refactored code when I reduced the number of variables and overlooked that. Fixed now